### PR TITLE
Create slides and samples for defer (#77)

### PR DESCRIPTION
**Pull request by linuxmonk**
* Describes what is defer
* How to use it
* How deferred functions are called (i.e. in reverse order)
* Illustrates usage with two samples

Slides: https://pr78-dot-gotraining-testing.appspot.com/gocourse-2019-05-melbourne.slide#66

### CODE REVIEW

File: samples/defer3.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+func f1() {
+	for i := 0; i < 3; i++ {
```
Comment (anzboi) [inline samples/defer3.go:7]: Can we add a one line description for `f1` and `f2`, to illustrate why they print in different order.
Remember to change the playspace to add the comments and update the link in the slides.

File: samples/defer3.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+func f2() {
+	for i := 0; i < 3; i++ {
+		defer func() {
+			fmt.Print(i, " ")
+		}()
+	}
+}
+
+func f3() (result int) {
+	defer func() {
```
Comment (anzboi) [inline samples/defer3.go:24]: Same here, a brief description of the behaviour here, try to keep to one line.

File: content/gocourse-2019-05-melbourne.slide [hunk 3]
```diff
@@ -815,50 +815,15 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
...
-- [[https://goplay.space/#pMbrba6y9cY][Defer]] with [[https://goplay.space/#pc68j1BPMxn][extended sample]]
-
+* Defer
+
+- `defer` delays the execution of a function until the surrounding function returns
+- When `defer` is called the function value and parameters to the call are evaluated and saved
+- Deferred functions are called when function returns or stack unwinds
+- Syntax:
+	defer <function call>
+- [[https://goplay.space/#Yne2XHUz3MH][Deferred functions are called]] in reverse order
```
Comment (anzboi) [inline content/gocourse-2019-05-melbourne.slide]: I'm gonna be picky here, This looks weird on the slides having 'Defferred functions are called' being the highlighted link.
IMO either make the entire line the link, or 'In reverse order'.

**Review by anzboi — CHANGES_REQUESTED** [review]
Looking good, I would like to change a couple of small things.

File: samples/defer3.go (hunk 1 above)
Comment (linuxmonk) [reply samples/defer3.go:7]: Done!

File: samples/defer3.go (hunk 2 above)
Comment (linuxmonk) [reply samples/defer3.go:24]: Done

File: content/gocourse-2019-05-melbourne.slide (hunk 3 above)
Comment (linuxmonk) [reply content/gocourse-2019-05-melbourne.slide]: Done! Picky is good. Iron out all the chinks

**Review by anzboi — APPROVED** [review]
LGTM

**Pull request merged**